src/components/progress-bar/progress-bar.scss

```diff
@@ -0,0 +1,205 @@
+@import "variables";
+@import "default-theme";
+
+$progress-bar-height: 5px !default;
+$progress-bar-full-animation-duration: 2s !default;
+$progress-bar-piece-animation-duration: 250ms !default;
+
+/** In buffer mode a repeated SVG object is used as a background.  Each of the following defines the SVG object for each
+    of the class defined colors.
+
+    Each string is a URL encoded version of:
+
+    <svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"
+         version="1.1" x="0px" y="0px" enable-background="new 0 0 5 2"
+         xml:space="preserve" viewBox="0 0 5 2" preserveAspectRatio="none slice">
+      <circle cx="1" cy="1" r="1" fill="{INJECTED_COLOR}"/>
+    </svg>
+
+    */
+$buffer-bubbles-primary: (
+  "data:image/svg+xml;charset=UTF-8,%3Csvg%20version%3D%271.1%27%20xmlns%3D%27http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%27" +
+  "%20xmlns%3Axlink%3D%27http%3A%2F%2Fwww.w3.org%2F1999%2Fxlink%27%20x%3D%270px%27%20y%3D%270px%27%20enable-backgroun" +
+  "d%3D%27new%200%200%205%202%27%20xml%3Aspace%3D%27preserve%27%20viewBox%3D%270%200%205%202%27%20preserveAspectRatio" +
+  "%3D%27none%20slice%27%3E%3Ccircle%20cx%3D%271%27%20cy%3D%271%27%20r%3D%271%27%20fill%3D%27" +
+  md-color($md-primary, 100) + "%27%2F%3E%3C%2Fsvg%3E") !default;
```

Comment:
That article is the reference for this. Used to reference it everytime this topic came.

@jelbourn do you know if it's possible to use something like LESS Hat (http://lesshat.madebysource.com/) but for SASS? Maybe we have to build our own functions... Basically it's a library that allows you to use and declare complex functions and end up with stuff like:

```
md-css-svg-optimize('
    <svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"
         version="1.1" x="0px" y="0px" enable-background="new 0 0 5 2"
         xml:space="preserve" viewBox="0 0 5 2" preserveAspectRatio="none slice">
      <circle cx="1" cy="1" r="1" fill="${md-color($md-primary, 100)}"/>
    </svg>');
```

directly in the code and then perform all the string operations necessary to inject and optimize the proper code. I'm pretty sure SCSS is that powerful, and it would help streamline our optimizations AS WELL AS declare the SVG natively inside the css. Would be a big PLUS!

Such an optimization is to remove, for example, the xmlns, since browsers don't need it for `image/svg+xml` mimetypes.